codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectEncoder.java

```diff
@@ -150,8 +150,11 @@ protected void encode(ChannelHandlerContext ctx, Object msg, List<Object> out) t
                         out.add(buf);
                     } else {
                         // Need to produce some output otherwise an
-                        // IllegalStateException will be thrown
-                        out.add(EMPTY_BUFFER);
+                        // IllegalStateException will be thrown as we did not write anything
+                        // Its ok to just write an EMPTY_BUFFER as if there are reference count issues these will be
+                        // propagated as the caller of the encode(...) method will release the original
+                        // buffer.
```

Comment:
also is this true: `writing an empty buffer will not actually write anything on the wire, so if there is a user error with msg it will not be visible externally`

Reply:
done